core/domain/stats_jobs_one_off.py

```diff
@@ -307,11 +307,18 @@ def reduce(exp_id, stringified_values):
 
             # Compute total_hit_count and first_hit_count for the states.
             for state_name in state_hit_counts_for_this_version:
-                state_stats_mapping[state_name].total_hit_count_v1 += (
-                    state_hit_counts_for_this_version[state_name][
-                        'total_hit_count'])
-                state_stats_mapping[state_name].first_hit_count_v1 += (
-                    len(state_session_ids_by_version[version][state_name]))
+                try:
+                    state_stats_mapping[state_name].total_hit_count_v1 += (
+                        state_hit_counts_for_this_version[state_name][
+                            'total_hit_count'])
+                    state_stats_mapping[state_name].first_hit_count_v1 += (
+                        len(state_session_ids_by_version[version][state_name]))
+                except KeyError:
+                    yield (
+                        'ERROR: State not in stats mapping exp_id %s, version '
+                        '%s, State %s, state_stats_mapping [%s]' % (
+                            exp_id, version, state_name,
+                            ', '.join(map(str, state_stats_mapping.keys()))))
```

Comment:
Should we do `raise` after the yield to make sure this still gets treated as an error?

Also you _might_ consider logging the whole of state_stats_mapping as a spot-check to see whether it looks reasonable, since we're logging stuff anyway.

Reply:
@seanlip Added the raise statement. We might not need to be logging the entire state_stats_mapping after what Tony suggested about encoding state names. Let's try that fix first to see if it fixes this issue.

Reply:
It's up to you, but note that, the more you log, the better. Otherwise you may need a subsequent run to gather more information.